### PR TITLE
[6866] Add bound checks for environment variables (4-3-stable)

### PR DESCRIPTION
**Pull request by Nishant2022**
Cherry pick from https://github.com/irods/irods/pull/7184

### CODE REVIEW

**Comment by korydraughn** [comment]
Conflicts found by github.

**Comment by Nishant2022** [comment]
Removed conflict

File: unit_tests/cmake/test_config/irods_environment_variables.cmake [hunk 1]
```diff
@@ -0,0 +1,7 @@
+set(IRODS_TEST_TARGET irods_environment_variables)
+
+set(IRODS_TEST_SOURCE_FILES ${CMAKE_CURRENT_SOURCE_DIR}/src/main.cpp
+                            ${CMAKE_CURRENT_SOURCE_DIR}/src/test_environment_variables.cpp)
+ 
+set(IRODS_TEST_LINK_LIBRARIES irods_common
+                              irods_server)
```
Comment (korydraughn) [inline unit_tests/cmake/test_config/irods_environment_variables.cmake:7]: Is `irods_server` needed?

Due to time, please open an issue about investigating this.

**Review by korydraughn — APPROVED** [review]
Please add the pound and resolve that comment.

**Review by alanking — APPROVED** [review]

**Pull request merged**